internal/servers/controller/handler.go

```diff
@@ -68,99 +68,70 @@ func (c *Controller) handler(props HandlerProperties) (http.Handler, error) {
 	callbackInterceptingHandler := wrapHandlerWithCallbackInterceptor(commonWrappedHandler, c)
 	printablePathCheckHandler := cleanhttp.PrintablePathCheckHandler(callbackInterceptingHandler, nil)
 	eventsHandler, err := common.WrapWithEventsHandler(printablePathCheckHandler, c.conf.Eventer, c.kms, props.ListenerConfig)
-	if err != nil {
-		return nil, err
-	}
 
-	return eventsHandler, nil
+	return eventsHandler, err
 }
 
-func handleGrpcGateway(c *Controller, props HandlerProperties) (http.Handler, error) {
-	// Register*ServiceHandlerServer methods ignore the passed in ctx. Using it
-	// now however in case this changes in the future.
-	ctx := props.CancelCtx
-	currentServices := c.gatewayServer.GetServiceInfo()
-	dialOptions := gatewayDialOptions(c.gatewayListener)
+func (c *Controller) registerGrpcServices(ctx context.Context, s *grpc.Server) (*grpc.Server, error) {
```

Comment:
TBD: We're receiving the gRPC Server `s` as a reference and then using it directly in the Register calls below. At the end, we're returning the object back to the caller. This isn't a great flow as it makes the caller think that what they call with (`s`) isn't going to be mutated, when in fact it is.

Solutions:
1. Don't return gRPC Server back. This makes it a bit clearer to the caller that we're mutating what they passed in, and because these are private functions, I think there should be the assumption that the caller should be aware of the inner workings of this function (to an extent).
2. Deep copy of `s`, and then use that into the Register calls and return that object. This is an interesting approach, but can be complex considering we're working with an object that already has state and has pointer fields within it.

Comments welcome here, thanks! 🙇

Reply:
Went with 1. Addressed in [c02c198](https://github.com/hashicorp/boundary/pull/1874/commits/c02c1980037a178a6bf6acd05b63ae11dc568d09)